ui/dashboards/src/views/ViewDashboard/ViewDashboard.tsx

```diff
@@ -58,7 +107,10 @@ export function ViewDashboard(props: ViewDashboardProps) {
           initialRefreshInterval={initialRefreshInterval}
           enabledURLParams={true}
         >
-          <TemplateVariableProvider initialVariableDefinitions={spec.variables}>
+          <TemplateVariableProvider
+            initialVariableDefinitions={spec.variables}
+            externalVariableDefinitions={externalVariableDefinitions}
```

Comment:
Can this prop be optional? It looks like if it’s empty the project and global vars form controls won’t show so that’s good but just want to call out that embed users may not have implemented the project and global var spec so this and how the requests are made should be opt-in (For example, our onSave logic is passed into ViewDashboard from the app which could be an approach for the GET request TODO above)

Reply:
(Really cool PR by the way! 🚀 Sorry if I jumped in to review too soon)

Reply:
Thanks! No problem, you're right. Don't hesitate to give feedback. There's still things that I will change though, like if I can simplify the store's statemap to prevent from things like that (state[source] || {})[name] which is completely ugly haha. I'm thinking of either a record with tuple [string,string|undefined] as key or a dedicated object. Not sure yet.

Reply:
Just kind of skimming this so I’d need to dig in a bit more but it might be possible to have it a bit more integrated with the variables Zustand store, which leverages immer produce

Reply:
Side note, @Gladorme, we could potentially start leveraging Zustand more in the rest of the app too as it gets more complex

Reply:
Note to myself: Don´t use tuple as Record key. As Macron said "This is huge mistake" :smile: 
https://www.typescriptlang.org/play?#code/MYewdgzgLgBA1gUwJ4EYBcMDa0BOBLMAcwBoZcDCYAfGAVzABMEAzAhBgXRgF4sAiMAEMAtghR9SfAAo4QAKwTAofDgG4AUKEixESAEwZsUfEVLki1OoxZtOPfkNHjJM+YuVr1m8BBAAbBAA6PxBCAApdFB5uXl09AEogA

@sjcobb If you don´t mind I would plan to create a dedicated MultiMap object. I see everywhere people using concatenation to create a single string key from multiple ones, but I'm not confident on which character will never be authorized in names.
Then I prefer to create something similar to this MultiMap object here https://stackoverflow.com/a/74594400/5690223 

Please shout if you disagree on that.

Reply:
@sjcobb I didn't touch too much at Zustand stuff, but if it can simplify contexts / store: it would be awesome 😄

Reply:
I just updated the PR, to I think simplify the state map. But it doesn´t work yet perfectly, 
I have the list variables stuck in loading state, I don´t know why. I'll need anyway some more time to check and fix.

![image](https://github.com/perses/perses/assets/10258608/5cd9b205-8f60-43d6-847c-b23e81cda321)

Reply:
## I found out my issue !
After 2 days of investigation, it ended up that I didn't read enough zustand documentation..... 
I have forgotten the ``[immerable] = true`` in my state class .....
Now it works like a charm with it :rocket: 

Now, to go back to the original comment. I ensured the fact that the ``externalVariableDefinitions`` can be indeed not be set. It still works the same without it. So @sjcobb I will put the conv as resolved.

Still I have some `TODO(Célian)` comments to resolve before moving out of DRAFT state.